components/DirectorySearch.tsx

```diff
@@ -0,0 +1,69 @@
+import { useRouter } from 'next/router';
+import { ChangeEvent, useEffect, useState } from 'react';
+import Club from '../models/club';
+
+const fetchResults = async (name: string) => {
+  const res = await fetch(`/api/club?name=${name}`);
+  const data = await res.json();
+  if (!data?.clubs || name === '') return [];
+  return data.clubs;
+};
+
+const DirectorySearch = () => {
```

Comment:
Is this component used only on the homepage, or also in the Directory search field? If it's only for the homepage it might make sense to have a more specific name.

Reply:
I think you're right because the other search bar will work a little bit different because of the grid